storage/database/db_manager.go

```diff
@@ -1436,6 +1441,109 @@ func (dbm *databaseManager) DeleteBlock(hash common.Hash, number uint64) {
 	dbm.cm.deleteBlockCache(hash)
 }
 
+const badBlockToKeep = 100
```

Comment:
Before the change, it has `maxBadBlocks` variable as a limitation, however, you changed to store bad blocks in db. Is this limitation still needed?

Reply:
Good suggestion. Lake also pointed out this part, and it is under discussion.